zkchannels-contract/zkchannel_smartpy_script.py

```diff
@@ -40,89 +39,113 @@ def is_g1_not_zero(val):
 
     def __init__(self, cid, custAddr, merchAddr, custPk, merchPk, custFunding, merchFunding, selfDelay, g2, merchPk0, merchPk1, merchPk2, merchPk3, merchPk4, merchPk5, close_flag):
         self.init(
-                  cid               = cid,
-                  custAddr          = custAddr,
-                  merchAddr         = merchAddr,
-                  custPk            = custPk,
-                  merchPk           = merchPk,
-                  custBal           = sp.mutez(0),
-                  merchBal          = sp.mutez(0),
-                  custFunding       = custFunding,
-                  merchFunding      = merchFunding,
-                  status            = sp.nat(AWAITING_FUNDING),
-                  revLock           = sp.bytes("0x00"),
-                  selfDelay         = selfDelay,
-                  delayExpiry       = sp.timestamp(0),
-                  g2                = g2,
-                  merchPk0          = merchPk0,
-                  merchPk1          = merchPk1,
-                  merchPk2          = merchPk2,
-                  merchPk3          = merchPk3,
-                  merchPk4          = merchPk4,
-                  merchPk5          = merchPk5,
+                  cid               = cid,              # the unique identifier for the channel.
+                  custAddr          = custAddr,         # customer tz1 address
+                  merchAddr         = merchAddr,        # merchant tz1 address
+                  custPk            = custPk,           # customer tezos public key
+                  merchPk           = merchPk,          # merchant tezos public key
+                  custBal           = sp.mutez(0),      # customer's closing balance
+                  merchBal          = sp.mutez(0),      # merchant's closing balance
+                  custFunding       = custFunding,      # customer's initial balance
+                  merchFunding      = merchFunding,     # merchant's initial balance
+                  status            = sp.nat(AWAITING_CUST_FUNDING), # contract status
+                  revLock           = sp.bytes("0x00"), # revLock initialized to 0x00
+                  selfDelay         = selfDelay,        # delay in seconds
```

Comment:
what is the effect of delay?

Reply:
added:
`# An enforced delay period that must have elapsed between calling custClose and custClaim, and between calling expiry and merchClaim.`